### DISCUSSION TITLE
Import into formatter settings

### DISCUSSION THREAD

**Discussion by disacrol** [checkstyle/eclipse-cs#527] [Q&A] [answered]
Hi,

eclipse-cs does provide full support for checkstyle rule sets to be imported and used.

But multiple checkstyle rules may impact formatting / naming conventions / etc.
These may clash with existing formatting settings in eclipse.
For example, the ones defined in Preferences | Java | Code Style | Formatter.
This can lead to eclipse automatic code formatting actually leading to new check style issues detected in the code.

Isn't there any way to import a checkstyle ruleset file into eclipse's formatter settings to avoid this?
Does one really need to go over formatting settings and manually make them match one by one with the rules from checkstyle?

Are there any plans for eclipse-cs to ever support such functionality?
IntelliJ IDEA natively supports this.

Thanks.

**Comment by romani** [comment, marked as answer]
Please step in project and make it happen, we need more maintainers and contributors.
